dimod/constrained.py

```diff
@@ -1090,6 +1090,29 @@ def num_biases(self) -> int:
                           for const in self.constraints.values())
         return num_biases
 
+    def num_quadratic_real_variables(self) -> int:
```

Comment:
what about ``num_variables_real_quadratic`` so it's the number of variables that are real and have quadratic biases? That way you have the "number of variables" first, then filtered by "real" then filtered by "quadratic". Might provide better order when there are lots of such functions.
Then the other would be ``num_linear_variables_real`` for the same considerations. 
Ugly, in both cases. 
At least it is not "continuous"

Reply:
The problem is I don't want to confuse it with `num_variables` which is unique variables in the CQM, and this which counts them twice if they appear in two constraints.

Reply:
right, but the counted object is variable, so you have to rely on the modifiers to distinguish

Reply:
I think I like `num_linear_biases_real` and `num_quadratic_variables_real`, where real is a narrowing down of existing fields `num_linear_biases` and `num_quadratic_variables`.